### PR TITLE
[Config] Add comment on array methods

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This PR adds comment section (including `@deprecated`, default, and examples) in the Generated PHP config for array's method headers.

### CODE REVIEW

File: src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php [hunk 1]
```diff
@@ -125,14 +126,19 @@ private function buildNode(NodeInterface $node, ClassBuilder $class, string $nam
 
     private function handleArrayNode(ArrayNode $node, ClassBuilder $class, string $namespace): void
     {
+        $comment = $this->getComment($node);
+        if ('' !== $comment) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php]: to be merged with the previous line

File: src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php [hunk 2]
```diff
@@ -220,16 +226,21 @@ public function NAME(string $VAR, TYPE $VALUE): static
         $this->classes[] = $childClass;
         $property = $class->addProperty($node->getName(), $childClass->getFqcn().'[]');
 
+        $comment = $this->getComment($node);
+        if ('' !== $comment) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php]: to be merged with the previous line

**Review by nicolas-grekas — APPROVED** [review]
:+1: once rebased, with minor comments

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jderusse.

**Pull request merged**